### PR TITLE
WIP: fix widget placement for input method

### PR DESCRIPTION
**Pull request by equalsraf**
Reported by @wsdjeg. @bfredl suggested calling setInputItemRectangle()
on cursor change would address this.

---

In truth I have no way to test this, crossing my fingers :)

### CODE REVIEW

**Comment by equalsraf** [comment]
Well. It seems this did not work.

**Comment by wsdjeg** [comment]
@equalsraf just a note, I get same issue in sublime text, but this package fix the issue, 

https://github.com/chikatoike/IMEsupport

**Comment by equalsraf** [comment]
Perhaps related to  https://bugreports.qt.io/browse/QTBUG-60334

**Comment by zhmars** [comment]
This patch seems to work for me, but I do not know much about it.

```cpp
--- neovim-qt-0.2.11/src/gui/shell.cpp  2019-01-05 12:12:44.738202578 +0800
+++ src/neovim-qt-0.2.11/src/gui/shell.cpp      2019-01-05 13:47:18.236589555 +0800
@@ -613,6 +613,7 @@
 {
     update(neovimCursorRect());
     m_cursor_pos = QPoint(col, row);
+    qApp->inputMethod()->update(Qt::ImCursorRectangle);
     update(neovimCursorRect());
 }

@@ -1193,7 +1194,7 @@
         return font();
     // } else if ( query == Qt::ImMicroFocus ) {
     } else if ( query == Qt::ImMicroFocus || query == Qt::ImCursorRectangle ) {
-        return QRect(neovimCursorTopLeft(), QSize(0, cellSize().height()));
+        return QRect(neovimCursorTopLeft(), cellSize());
     }

     return QVariant();
```

**Comment by equalsraf** [comment]
Strange, in my system  it has no effect. @zhmars I've removed the original patch from this PR and kept the cellSize change, does this work for you?

**Comment by zhmars** [comment]
Sadly, it does not work. I still need to call `update(Qt::ImCursorRectangle)`.

**Comment by equalsraf** [comment]
Ok, I've added the change, confirmed it works on my system. Will merge as soon as the checks are clear.

File: src/gui/shell.cpp [hunk 1]
```diff
@@ -614,6 +614,7 @@ void Shell::setNeovimCursor(quint64 row, quint64 col)
 	update(neovimCursorRect());
 	m_cursor_pos = QPoint(col, row);
 	update(neovimCursorRect());
+	qApp->inputMethod()->update(Qt::ImCursorRectangle);
```
[the remark this answers is not in the document]
Comment (equalsraf) [reply src/gui/shell.cpp]: Pushed a fix into master https://github.com/equalsraf/neovim-qt/pull/497. Thanks @zhmars

[the remark this answers is not in the document]
Comment (zhmars) [reply src/gui/shell.cpp]: Great, can we have nightly build ? I can help to test it on windows.

[the remark this answers is not in the document]
Comment (equalsraf) [reply src/gui/shell.cpp]: Tagged as **staging**, if the CI does not fail there should be a windows binary in the releases page shortly.

**Pull request merged**